### ISSUE TITLE
Server not Starting Centos 7 

### ISSUE THREAD

**Issue by adamwbb** [hexparrot/mineos-node#153]
(moved from mineos python version) link to original issue thread: https://github.com/hexparrot/mineos/issues/131

Server will not start when created as a non-root user.

this is what i get from the log files when i click start
```
{"level":"info","message":"[test1] received request \"modify_sc\"","timestamp":"2015-12-04T16:50:59.494Z"}
{"level":"info","message":"[test1] 69.133.202.230 issued command : \"start\"","timestamp":"2015-12-04T16:50:59.663Z"}
{"level":"info","message":"[test1] received request \"start\"","timestamp":"2015-12-04T16:50:59.663Z"}
{"command":"start","uuid":"322c05f0-9aa7-11e5-aca0-6b3c2441fbf6","time_initiated":1449247859663,"success":false,"err":1,"time_resolved":1449247859876,"level":"error","message":"[test1] command \"start\" errored out:","timestamp":"2015-12-04T16:50:59.877Z"}
```

I have all dependencies installed.
rdiff-sync
rdiff-backup
nodejs
java
screen

**Comment by adamwbb** [comment]
dont know if this information is relavent but here is the log when starting the the mineos service
```
[root@hexihostn1 minecraft]# node webui.js
base_directory found in mineos.conf, using: /var/games/minecraft
info: [test] Discovered server
error: [test] Create tail on logs/latest.log failed
info: [test] Watching for file generation: logs/latest.log
error: [test] Create tail on server.log failed
info: [test] Watching for file generation: server.log
error: [test] Create tail on proxy.log.0 failed
info: [test] Watching for file generation: proxy.log.0
MineOS webui listening on HTTPS://0.0.0.0:8443
info: Starting up server, using commit: 48c7387 another server detection rewrite
```

**Comment by hexparrot** [comment]
Alright, we'll check it the long way then.   This time, as `root`:

    root@core ~# cd /usr/games/minecraft/
    root@core games/minecraft# node mineos_console.js -s adsf get_start_args
    [adsf] Successfully executed "get_start_args"
    [ '-dmS',
      'mc-adsf',
      '/usr/bin/java',
      '-server',
      '-Xmx256M',
      '-Xms256M',
      '-jar',
      'minecraft_server.1.8.8.jar',
      'nogui' ]
    root@core games/minecraft# screen -dmS mc-adsf /usr/bin/java -server -Xmx256M -Xms256M -jar minecraft_server.1.8.8.jar nogui

This way it's actually testing that screen is present as well as whether the java arguments are causing the problem.

**Comment by adamwbb** [comment]
ok here is what i have
```
[root@hexihostn1 minecraft]# cd /usr/games/minecraft/
[root@hexihostn1 minecraft]# node mineos_console.js -s test get_start_args
[test] Successfully executed "get_start_args"
[ '-dmS',
  'mc-test',
  '/usr/bin/java',
  '-server',
  '-Xmx1024M',
  '-Xms1024M',
  '-jar',
  'spigot-1.8.8.jar',
  'nogui' ]
[root@hexihostn1 minecraft]# screen -dmS mc-test /usr/bin/java -server -Xmx1024M -Xms1024M -jar minecraft_server.1.8.8.jar nogui
[root@hexihostn1 minecraft]#
```
should there be any other output?

**Comment by hexparrot** [comment]
Did the server start, then?

Do you see the process open in `top`?

You can connect to the server's console screen directly now with "screen -r" (as the `mc` user, not `root`).

**Comment by adamwbb** [comment]
nothing.
no server process showing in ```top```
and i get there is no screen to be resumed when i run ```screen -r``` as ```mc```

**Comment by hexparrot** [comment]
Alright then this AS `mc`:

    cd /var/games/minecraft/test
    /usr/bin/java -server -Xmx1024M -Xms1024M -jar minecraft_server.1.8.8.jar nogui

**Comment by adamwbb** [comment]
ok so it starts in the command line with that command however the dashboard reports it as not started and unable to send any commands via the mineos panel log console it throws error saying server needs to be running top perform this command.

output on ssh shell:
.```
[mc@hexihostn1 test]$ /usr/bin/java -server -Xmx1024M -Xms1024M -jar minecraft_server.1.8.8.jar nogui
[16:57:50] [Server thread/INFO]: Starting minecraft server version 1.8.8
[16:57:50] [Server thread/INFO]: Loading properties
[16:57:50] [Server thread/WARN]: Failed to load eula.txt
[16:57:50] [Server thread/INFO]: You need to agree to the EULA in order to run the server. Go to eula.txt for more info.
[16:57:50] [Server thread/INFO]: Stopping server
[16:57:50] [Server Shutdown Thread/INFO]: Stopping server
[mc@hexihostn1 test]$ /usr/bin/java -server -Xmx1024M -Xms1024M -jar minecraft_server.1.8.8.jar nogui
[16:58:14] [Server thread/INFO]: Starting minecraft server version 1.8.8
[16:58:14] [Server thread/INFO]: Loading properties
[16:58:14] [Server thread/INFO]: Default game type: CREATIVE
[16:58:14] [Server thread/INFO]: Generating keypair
[16:58:14] [Server thread/INFO]: Starting Minecraft server on 0.0.0.0:25581
[16:58:14] [Server thread/INFO]: Using epoll channel type
[16:58:14] [Server thread/WARN]: Failed to load user banlist:
java.io.FileNotFoundException: banned-players.json (No such file or directory)
        at java.io.FileInputStream.open0(Native Method) ~[?:1.8.0_65]
        at java.io.FileInputStream.open(FileInputStream.java:195) ~[?:1.8.0_65]
        at java.io.FileInputStream.<init>(FileInputStream.java:138) ~[?:1.8.0_65]
        at com.google.common.io.Files.newReader(Files.java:86) ~[minecraft_server.1.8.8.jar:?]
        at mb.g(SourceFile:124) ~[minecraft_server.1.8.8.jar:?]
        at kn.z(SourceFile:99) [minecraft_server.1.8.8.jar:?]
        at kn.<init>(SourceFile:25) [minecraft_server.1.8.8.jar:?]
        at ko.i(SourceFile:172) [minecraft_server.1.8.8.jar:?]
        at net.minecraft.server.MinecraftServer.run(SourceFile:421) [minecraft_server.1.8.8.jar:?]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_65]
[16:58:14] [Server thread/WARN]: Failed to load ip banlist:
java.io.FileNotFoundException: banned-ips.json (No such file or directory)
        at java.io.FileInputStream.open0(Native Method) ~[?:1.8.0_65]
        at java.io.FileInputStream.open(FileInputStream.java:195) ~[?:1.8.0_65]
        at java.io.FileInputStream.<init>(FileInputStream.java:138) ~[?:1.8.0_65]
        at com.google.common.io.Files.newReader(Files.java:86) ~[minecraft_server.1.8.8.jar:?]
        at mb.g(SourceFile:124) ~[minecraft_server.1.8.8.jar:?]
        at kn.y(SourceFile:91) [minecraft_server.1.8.8.jar:?]
        at kn.<init>(SourceFile:27) [minecraft_server.1.8.8.jar:?]
        at ko.i(SourceFile:172) [minecraft_server.1.8.8.jar:?]
        at net.minecraft.server.MinecraftServer.run(SourceFile:421) [minecraft_server.1.8.8.jar:?]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_65]
[16:58:14] [Server thread/WARN]: Failed to load operators list:
java.io.FileNotFoundException: ops.json (No such file or directory)
        at java.io.FileInputStream.open0(Native Method) ~[?:1.8.0_65]
        at java.io.FileInputStream.open(FileInputStream.java:195) ~[?:1.8.0_65]
        at java.io.FileInputStream.<init>(FileInputStream.java:138) ~[?:1.8.0_65]
        at com.google.common.io.Files.newReader(Files.java:86) ~[minecraft_server.1.8.8.jar:?]
        at mb.g(SourceFile:124) ~[minecraft_server.1.8.8.jar:?]
        at kn.A(SourceFile:107) [minecraft_server.1.8.8.jar:?]
        at kn.<init>(SourceFile:29) [minecraft_server.1.8.8.jar:?]
        at ko.i(SourceFile:172) [minecraft_server.1.8.8.jar:?]
        at net.minecraft.server.MinecraftServer.run(SourceFile:421) [minecraft_server.1.8.8.jar:?]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_65]
[16:58:14] [Server thread/WARN]: Failed to load white-list:
java.io.FileNotFoundException: whitelist.json (No such file or directory)
        at java.io.FileInputStream.open0(Native Method) ~[?:1.8.0_65]
        at java.io.FileInputStream.open(FileInputStream.java:195) ~[?:1.8.0_65]
        at java.io.FileInputStream.<init>(FileInputStream.java:138) ~[?:1.8.0_65]
        at com.google.common.io.Files.newReader(Files.java:86) ~[minecraft_server.1.8.8.jar:?]
        at mb.g(SourceFile:124) ~[minecraft_server.1.8.8.jar:?]
        at kn.C(SourceFile:123) [minecraft_server.1.8.8.jar:?]
        at kn.<init>(SourceFile:30) [minecraft_server.1.8.8.jar:?]
        at ko.i(SourceFile:172) [minecraft_server.1.8.8.jar:?]
        at net.minecraft.server.MinecraftServer.run(SourceFile:421) [minecraft_server.1.8.8.jar:?]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_65]
[16:58:14] [Server thread/INFO]: Preparing level "world"
[16:58:14] [Server thread/INFO]: Preparing start region for level 0
[16:58:15] [Server thread/INFO]: Preparing spawn area: 11%
[16:58:16] [Server thread/INFO]: Preparing spawn area: 23%
[16:58:17] [Server thread/INFO]: Preparing spawn area: 38%
[16:58:18] [Server thread/INFO]: Preparing spawn area: 60%
[16:58:19] [Server thread/INFO]: Preparing spawn area: 82%
[16:58:20] [Server thread/INFO]: Done (6.024s)! For help, type "help" or "?"
[16:58:20] [Server thread/INFO]: Starting GS4 status listener
[16:58:20] [Server thread/INFO]: Setting default query port to 25581
[16:58:20] [Query Listener #1/INFO]: Query running on 0.0.0.0:25581
```

**Comment by hexparrot** [comment]
Wait, so what exactly is the problem?

For starters, when `screen` isn't used, you shouldn't expect the webui to acknowledge the open server.  The webui checks for the instance of screen, which opens up java.

So if you want the webui to show the server, you'll have to stop the server and return to the original command (the one with `screen -dmS`).  The next point is each attempted start failed because the EULA was missing.  It was starting the server then stopping the process because the EULA wasn't accepted.

So at this point, you should be able to do the following:

a) stop the server with `stop`.  you can type this directly into the Minecraft instance.
alternate a) just kill the minecraft instance, doesn't matter how.

b) start the server with the webui.

**Comment by adamwbb** [comment]
whenever i try to start the server from the WebUI
this is what i get
![image](https://cloud.githubusercontent.com/assets/2437446/11610757/b4490614-9b79-11e5-88d9-3b0d2e57ab89.png)

**Comment by hexparrot** [comment]
Can you verify that no files in `/var/games/minecraft/servers/test` are owned by `root:root`?  Honestly, I have no clue at this point how to guide you to find the problem at hand.

Maybe if you want me to try logging in .

**Comment by adamwbb** [comment]
all files and folders in /var/games/minecraft/servers/test (including the test directory itself) are owned by mc:mc
![image](https://cloud.githubusercontent.com/assets/2437446/11610820/546e1ea2-9b7c-11e5-996c-9ec19daa0cef.png)

**Comment by adamwbb** [comment]
you can login and see the error for your self if you want:
https://192.99.3.48:8443
user: mc
pass: minecraft

**Comment by adamwbb** [comment]
does this mineos log output tell you anything?
``` {"command":"start","uuid":"26b1ac10-9ba7-11e5-b98b-03d91f0ac5a0","time_initiated":1449357791569,"success":false,"err":1,"time_resolved":1449357791780,"level":"error","message":"[test] command \"start\" errored out:","timestamp":"2015-12-05T23:23:11.780Z"} ```

**Comment by hexparrot** [comment]
There's still a screen instance running as root, which needs to be killed.  We can move this to email, wdchromium@gmail.com.

Or reboot the server.

**Comment by adamwbb** [comment]
i tried rebooting and now it goes back to before where you click start and nothing happens. not even a error notification.

**Comment by hexparrot** [comment]
Alright, figured it out.  Not related to MineOS, but instead `screen`.

    [mc@hexihostn1 asdf]$ screen
    [screen is terminating]
    [mc@hexihostn1 asdf]$ screen -S testsesh /usr/bin/java
    [screen is terminating]

Related issue: http://superuser.com/questions/781591/screen-is-terminating-for-non-root

Looks like the official CentOS7 issue is this: https://bugs.centos.org/view.php?id=7395

**Comment by adamwbb** [comment]
ok here is the bug fix for reference for those using CentOS 7

as ```root``` us the following commands:
```chmod u+s /usr/bin/screen
chmod 755 /var/run/screen```

**Comment by hexparrot** [comment]
Everything working as intended with that?

**Comment by adamwbb** [comment]
yes.

**Comment by Dekwin** [comment]
CentOS Linux release 7.3.1611 (Core). Same issue. 
`chmod u+s /usr/bin/screen
chmod 755 /var/run/screen`
I have to do it after each server reboot. 
How to set this rule permanently?

**Comment by hexparrot** [comment]
> How to set this rule permanently?

If it is unset after each reboot, it could be CentOS resets it (as per their own deliberate policy).  You can, if you choose, have a startup script re-do these intended chmodding; see documentation for CentOS for any specific files that may be used for this, but `/etc/rc.d/rc.local` is a good first candidate to look up.

**Comment by Dekwin** [comment]
@hexparrot Thanks for reply. I have another issue.
I installed server using tutorial [https://minecraft.codeemo.com/mineoswiki/index.php?title=MineOS-node_(yum)
](url)

I used the command(as root) systemctl start mineos to start server
It started, I signedIn into my server, but I didn't able to manage my servers(the server list was empty) and I didn't see my username at the right top corner.(I attached the screenshot)
![2017-01-12 14 10 38](https://cloud.githubusercontent.com/assets/13164807/21889374/4e37b99c-d8d1-11e6-969e-c04fc0a96230.png)
)

When I run the command "node service.js start" from /usr/games/minecraft folder (as root) the server starts and works properly. 
What is the reason of this problem?

I think I have to set chmod for some files?
